internal/service/service_test.go

```diff
@@ -0,0 +1,170 @@
+package service_test
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/gopheryan/jobby/internal/service"
+	"github.com/gopheryan/jobby/internal/testutils"
+	"github.com/gopheryan/jobby/jobmanagerpb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const echoPathRelative = "../../testdata/testprograms/echo"
+
+type mockUserGetter struct {
+	user string
+}
+
+func (m *mockUserGetter) GetUserContext(_ context.Context) string {
+	return m.user
+}
+
+// Unary calls are easy to test!
+// Just call them directly
+func TestUnaryCalls(t *testing.T) {
+	ctx := context.Background()
+	mockUserGetter := &mockUserGetter{user: "someuser"}
+	jobService := service.NewJobService(1, mockUserGetter, "/tmp")
+
+	t.Run("start-stop-status", func(tt *testing.T) {
+		resp, err := jobService.StartJob(ctx, &jobmanagerpb.StartJobRequest{
+			Command: echoPathRelative,
+			Args:    []string{"echo", "5"},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.NotNil(t, resp.JobId)
+
+		stopResp, err := jobService.StopJob(ctx, &jobmanagerpb.StopJobRequest{
+			JobId: resp.JobId,
+		})
+		require.NoError(t, err)
+		require.NotNil(t, stopResp)
+
+		time.Sleep(1 * time.Second)
+
+		statusResp, err := jobService.GetStatus(ctx, &jobmanagerpb.GetStatusRequest{
+			JobId: resp.JobId,
+		})
+		require.NoError(t, err)
+		require.NotNil(t, statusResp)
+		require.Equal(t, jobmanagerpb.Status_STATUS_STOPPED, statusResp.CurrentStatus)
+	})
+
+	t.Run("invalid-user", func(tt *testing.T) {
+		// Create a job
+		resp, err := jobService.StartJob(ctx, &jobmanagerpb.StartJobRequest{
+			Command: echoPathRelative,
+			Args:    []string{"echo", "5"},
+		})
+		require.NoError(tt, err)
+		require.NotNil(tt, resp)
+		require.NotNil(tt, resp.JobId)
+
+		// Change the user
+		mockUserGetter.user = "anotheruser"
+
+		// Stopping the job should fail with "not found" since we are no longer
+		// the right user
+		stopResp, err := jobService.StopJob(ctx, &jobmanagerpb.StopJobRequest{
+			JobId: resp.JobId,
+		})
+		require.Error(tt, err)
+		stat, ok := status.FromError(err)
+		require.True(tt, ok)
+		require.Equal(tt, codes.NotFound, stat.Code())
+		require.Nil(t, stopResp)
+	})
+
+}
+
+// Streaming is a little more challenging
+// We could generate some mocks (I like github.com/maxbrunsfeld/counterfeiter)
+// But for basic black box tests, a local server is easy enough to spin up
+func TestService(t *testing.T) {
+	srv := testutils.GrpcLocalServer{}
+	jobService := service.NewJobService(1, &mockUserGetter{user: "someuser"}, "/tmp")
```

Comment:
A little more future proof:
```suggestion
	jobService := service.NewJobService(1, &mockUserGetter{user: "someuser"}, os.TempDir())
```